src/app/add-clients/helperFunctions.tsx

```diff
@@ -0,0 +1,254 @@
+        } else {
+            newValue[personIndex].quantity = -1;
+        }
+        fieldSetter("adults", newValue);
+
+        const invalidAdultEntry = newValue.filter((value) => value.quantity === -1);
+        const nonZeroAdultEntry = newValue.filter((value) => value.quantity! > 0);
+        let newMessage = "";
+        if (invalidAdultEntry.length > 0) {
+            newMessage = "Please enter a valid entry.";
+        } else if (nonZeroAdultEntry.length === 0) {
+            newMessage = "This is a required field.";
+        }
+        errorSetter("adults", newMessage);
+    };
+};
+
+export const getChild = (
+    fieldSetter: SetFieldType,
+    children: People[],
+    index: number,
+    subFieldName: "personType" | "age"
+): OnChangeType => {
+    return (event) => {
+        const input = event.target.value;
+        if (subFieldName === "personType") {
+            children[index][subFieldName] = (
+                input !== "don't know" ? input : "child"
+            ) as PersonType;
+        } else {
+            children[index][subFieldName] = input !== "don't know" ? parseInt(input) : null;
+        }
+        fieldSetter("children", [...children]);
+    };
+};
+
+export const getBaby = (fieldSetter: SetFieldType, errorSetter: SetErrorType): OnChangeType => {
+    return (event) => {
+        const input = event.target.value;
+        if (input === "Yes") {
+            errorSetter("babyProducts", "N/A");
+            fieldSetter("babyProducts", true);
+        } else {
+            errorSetter("babyProducts", "");
+            if (input === "No") {
+                fieldSetter("babyProducts", false);
+            } else {
+                fieldSetter("babyProducts", null);
+            }
+        }
+    };
+};
+
+export const checkErrorOnSubmit = (
+    errorMessages: ErrorMessages,
+    errorSetter: React.Dispatch<React.SetStateAction<ErrorMessages>>
+): boolean => {
+    let errorExists = false;
+    let amendedErrorMessages = { ...errorMessages };
+    for (const [errorKey, errorMessage] of Object.entries(errorMessages)) {
+        if (errorMessage !== "") {
+            errorExists = true;
+        }
+        if (errorMessage === "N/A") {
+            amendedErrorMessages = {
+                ...amendedErrorMessages,
+                [errorKey]: "This is a required field.",
+            };
+        }
+    }
+    if (errorExists) {
+        errorSetter({ ...amendedErrorMessages });
+    }
+    return errorExists;
+};
+
+export const insertFamily = async (peopleArray: People[], familyID: string): Promise<boolean> => {
+    const familyRecords = peopleArray.map((person): FamilyDatabaseRecord => {
+        return {
+            family_id: familyID,
+            person_type: person.personType,
+            quantity: person.quantity ?? 1,
+            age: person.age ?? null,
+        };
+    });
+
+    const { error: error } = await supabase.from("families").insert(familyRecords);
+    return error === null;
```

Comment:
If you think it would improve the user experience try and propagate errors up and use a snackbar to show it to the user in a snackbar so they know what's going on.

Reply:
I think the current error text is sufficient. Thanks for the suggestion anyways :)